### DISCUSSION TITLE
Popover: Add open and onChange props . Too difficult to use without the button

### DISCUSSION THREAD

**Discussion by gregg-cbs** [tailwindlabs/headlessui#2360] [Ideas]
With the Popover component its entire functionality seems focused around the Popover.Button being used. I have cases where the place I want to use the popover is under an input and it should only show based on some criteria like input value not being empty and results being present.

This functionality should be as simple as:
`<Popover open={isOpen} onChange={(bool)=> setIsOpen(bool)>`

But instead now we have to wire everything up manually and its messy and tedious.

With my suggestion the outside click will still close the popover and the developer can control when the popover should be open or closed - best of both.

**Comment by gregg-cbs** [comment]
Duplicate of: https://github.com/tailwindlabs/headlessui/discussions/2206

**Comment by mwawrusch** [comment]
Yeah, running into the same problem. We have an search input box that should trigger a popover.
